ddtrace/ddtrace.go

```diff
@@ -79,6 +79,17 @@ type SpanContext interface {
 	ForeachBaggageItem(handler func(k, v string) bool)
 }
 
+// ReadWriteSpan implementations are spans with read/write access.
+type ReadWriteSpan interface {
+	Span
+
+	// Tag returns the tag value held by the given key.
+	Tag(key string) interface{}
+
+	// IsError reports wether the span is an error.
+	IsError() bool
+}
```

Comment:
Just a general question: how do we know that `Tag` and `IsError` are the only two pieces of information someone would need to know about the span?

Reply:
Currently, you can retrieve the following from a span using the `ReadWriteSpan` interface:
From `Tag`:
- Name     
- Service  
- Resource 
- Type 
- Any tag in Meta     
- Any tag in Metrics  

From `IsError`:
- If the span is an error. (Error message, type, stack and details can be retrieved from Meta using `Tag` above)

The only fields that you can't retrieve are `Start` and `Duration`. 
For `Start`, I don't believe this would be needed to make a decision on wether to drop a trace, and we decided to remove access to `Duration` [here](https://github.com/DataDog/dd-trace-go/pull/1381#discussion_r925601076) for the same reason.

For `Tag`, I agree that it may be not be clear that some fields (e.g. service, resource) can be accessed through this method, and customer's may think they can just access `Meta` and `Metrics`. Perhaps we could link the [ext package](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.40.1/ddtrace/ext) in the description so customers are aware they can read all these fields from the `Tag` method ?

Reply:
Good point about customer's not fully understanding what they can do with this code. Maybe an example would be good then (meaning a Go Example that will be in context with the godoc in pkg.go.dev)

Reply:
Sounds good, will work on adding an example!

Reply:
Added an [example](https://github.com/DataDog/dd-trace-go/pull/1381/commits/a837624b00e29f668d42c2e1ea9fc2af0d037176) which goes over different possibilities. Let me know what you think ?

Reply:
Example looks great, thanks!